WordPress/openverse
# Add point_alias DAG and add alias params to create_new_es_index DAGs

## Fixes



Fixes #3493 by @stacimc

## Description



This PR adds DAGs for pointing a `target_alias` at an existing elasticsearch index, with one DAG each for staging and production. Before the alias is applied, it will first be removed from the index to which it currently points (if applicable). There is also an option to delete this pre-existing index if desired. This was done by slightly extending the existing `point_alias` taskgroup from the `create_proportional_by_source_staging_index` DAG.

The `point_alias` task group is then used in a separate `point_<environment>_alias` DAG factory, and also added as a step to the `create_new_<environment>_es_index` DAGs to add functionality to those DAGs as well.

## Testing Instructions


### Test that `create_proportional_by_source_staging_index` DAGs still work

1. Run the DAG locally with default params and verify it all passes. In elasticvue you should see a new index with a name like `audio-50-percent-proportional-20240308t235423` and the alias` audio-subset-by-source`. 
2. Run the DAG a second time and verify all passes. This time `audio-subset-by-source` should have been *removed* from the index created in the previous step, and applied to a new index with a later timestamp.

### Test the new DAG

1. Run `point_staging_alias` locally. Pass in an existing index and an alias that does not currently exist. Verify that the alias gets added to the selected index. `remove_alias_from_existing_index` and `delete_old_index` steps will be skipped.
2. Run again using the same alias, pointed to a different index. This time `remove_alias_from_existing_index` will not be skipped; the alias should be removed from the old index before being applied to the new one. `delete_old_index` is still skipped.
3. Run again using the same alias and another index but also enable `should_delete_old_index`. This time no tasks should be skipped, and the alias should not only be moved to the new index but the old index should be deleted entirely.

### Test create_new_<x>_es_index DAGs

1. Run `create_new_staging_es_index` and do NOT pass in a target_alias. It should pass, and all the `point_alias` steps should be skipped.
2. Run again, but this time pass in an alias that does not already exist. Verify that it is added to your new index, and `remove_alias_from_existing_index` and `delete_old_index` steps are skipped
3. Run again with the same alias. Verify that it is added to your new index and removed from the index created in the previous step.
4. Run again with the same alias, and `should_delete_old_index` enabled. Verify it is added to the new index, and the index created in step 3 is deleted.


## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
